system-tests/lib/cre/contracts/contracts.go

```diff
@@ -141,6 +142,114 @@ func (d *dons) allDonCapabilities() []keystone_changeset.DonCapabilities {
+	capabilities := make([]capabilities_registry_v2.CapabilitiesRegistryCapability, 0)
+	donParams := make([]capabilities_registry_v2.CapabilitiesRegistryNewDONParams, 0)
+
+	// Collect unique capabilities and NOPs
+	capabilityMap := make(map[string]capabilities_registry_v2.CapabilitiesRegistryCapability)
+	nopMap := make(map[string]capabilities_registry_v2.CapabilitiesRegistryNodeOperator)
+
+	for _, don := range d.donsOrderedByID() {
+		// Extract capabilities
+		for _, cap := range don.Capabilities {
+			capID := fmt.Sprintf("%s@%s", cap.Capability.LabelledName, cap.Capability.Version)
+			if _, exists := capabilityMap[capID]; !exists {
+				capabilityMap[capID] = capabilities_registry_v2.CapabilitiesRegistryCapability{
+					CapabilityId:          capID,
+					ConfigurationContract: common.Address{},
+					Metadata:              []byte("{}"),
+				}
+			}
+		}
+
+		// Extract NOPs and nodes
+		for _, nop := range don.Nops {
+			nopName := nop.Name
+			if _, exists := nopMap[nopName]; !exists {
+				nopMap[nopName] = capabilities_registry_v2.CapabilitiesRegistryNodeOperator{
+					Admin: common.Address{}, // Will be set by the deployment framework
+					Name:  nopName,
+				}
+
+				// Add nodes for this NOP
+				for _, nodeID := range nop.Nodes {
+					peerID, err := p2pkey.MakePeerID(nodeID)
+					if err != nil {
+						continue // Skip invalid peer IDs
+					}
+					// Safe conversion: len(nops) is controlled and small
+					nodeOperatorID := uint32(len(nops)) //nolint:gosec // G115: len(nops) is small and controlled
+					nodes = append(nodes, capabilities_registry_v2.CapabilitiesRegistryNodeParams{
+						NodeOperatorId:      nodeOperatorID,
+						P2pId:               peerID,
+						Signer:              [32]byte{}, // Will be set by the deployment framework
+						EncryptionPublicKey: [32]byte{}, // Will be set by the deployment framework
+					})
+				}
+			}
+		}
+
+		// Create DON parameters
+		var capConfigs []capabilities_registry_v2.CapabilitiesRegistryCapabilityConfiguration
+		for _, cap := range don.Capabilities {
+			capID := fmt.Sprintf("%s@%s", cap.Capability.LabelledName, cap.Capability.Version)
+			configBytes := []byte("{}")
+			if cap.Config != nil {
+				// Convert proto config to bytes if needed
+				if protoBytes, err := proto.Marshal(cap.Config); err == nil {
+					configBytes = protoBytes
+				}
+			}
+			capConfigs = append(capConfigs, capabilities_registry_v2.CapabilitiesRegistryCapabilityConfiguration{
+				CapabilityId: capID,
+				Config:       configBytes,
+			})
+		}
+
+		var donNodes [][32]byte
+		for _, nop := range don.Nops {
+			for _, nodeID := range nop.Nodes {
+				peerID, err := p2pkey.MakePeerID(nodeID)
+				if err != nil {
+					continue
+				}
+				donNodes = append(donNodes, peerID)
+			}
+		}
+
+		donParams = append(donParams, capabilities_registry_v2.CapabilitiesRegistryNewDONParams{
+			Name:                     don.Name,
+			DonFamilies:              []string{}, // Default empty
```

Comment:
we should set a default, I think it's needed for V2.  some const in cre/environment/config package could be good `test-don-family`